build/xwalk.gni

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2016 Intel Corporation. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# This file contains Crosswalk-feature-related build flags. Users should
+# import it into their build folder by run:
+#   echo 'import("//xwalk/build/xwalk.gni")' > out/your_folder/args.gn'
+# Then run:
+#   gn gen out/your_folder
+
+# Set it false when building content_shell.
+declare_args() {
+  is_build_crosswalk = true
+}
+
+if (is_build_crosswalk) {
+  target_os = "linux"
```

Comment:
> My understanding: User should set target_os/target_cpu/is_debug by"gn args". Now we use this "echo import(xwalk.gni)" approach, User just need to edit this xwalk.gni to modify the values of this 3 lines.

Requiring users to modify `xwalk.gni` makes **zero** sense. Not only does this mean a file that is part of the checkout will need to be modified by each user, but it will also prevent multiple build directories from having different build flags.

If one wants to change `is_debug`, it's just a matter of adding it to `args.gn` via `gn args`, the `import` clause is irrelevant in this context:
```
import("//xwalk/build/xwalk.gni")
is_debug = false
```

The problem with the conditionals is real though, and I wasn't aware of it. For now, what comes to mind is having all the common settings in, say, `xwalk.gni` or `common.gni` and then separate files such as `linux.gni` and `windows.gni` with additional settings. `linux.gni` imports `xwalk.gni` and adds some settings after that, and users import `linux.gni` in their `args.gn` instead.

However, this approach is not very flexible, so I'll ask the gn-dev mailing list about this issue.